sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -317,7 +317,7 @@ nestedFunction
     ;
 
 nestedField
-    : ID DOT ID (DOT ID)*
+    : ident DOT ident (DOT ident)*
```

Comment:
Are you adding some fixes or only tests?
BTW one of `ident` options is `DOT? ID`. This parser rule accepts `field..subfield` I suspect.

Reply:
Good point, maybe we should add a `nestedIdent` rule.

Reply:
I think it can be part of this PR because it is related to one of the tests not parsing for fields with opensearch function names. For example, `dayOfWeek`.